mne/beamformer/_compute_beamformer.py

```diff
@@ -247,3 +250,187 @@ def _prepare_beamformer_input(info, forward, label, picks, pick_ori,
                          'values are: "dipole", "vertex" or None.')
 
     return is_free_ori, ch_names, proj, vertno, G
+
+
+def _compute_beamformer(method, G, Cm, reg, n_orient, weight_norm,
+                        pick_ori, reduce_rank, rank, is_free_ori,
+                        inversion=None):
+    """Compute a spatial filter (LCMV or DICS)."""
+    # Tikhonov regularization using reg parameter d to control for
+    # trade-off between spatial resolution and noise sensitivity
+    if method == 'lcmv':
+        Cm_inv, d = _reg_pinv(Cm.copy(), reg)
+    elif method == 'dics':
+        Cm_inv, _ = _reg_pinv(Cm, reg, rcond='auto')
+
+    if weight_norm is not None and inversion is not 'single':
+        # Compute square of Cm_inv used for weight normalization
+        Cm_inv_sq = np.dot(Cm_inv, Cm_inv)
+
+        if weight_norm == 'nai':
+            # estimate noise level based on covariance matrix, taking the
+            # smallest eigenvalue that is not zero
+            noise, _ = linalg.eigh(Cm)
+            if rank is not None:
+                rank_Cm = rank
```

Comment:
@britta-wstnr Is this correct? When `rank is not None`, the `noise` variable is a list and when `rank is None` the `noise` variable is a single value...

Reply:
and `rank_Cm` is not used anywhere when `rank is not None`.